CONTRIBUTING.md

```diff
@@ -2,7 +2,124 @@
 
 <a name="react-client-templates"/>
 
-### Code Generation Templates
+### Generators
+
+#### Basics and Terminology
+
+`packages/jmix-front-generator` contains the source code for `@haulmont/jmix-front-generator` library which is used for code generation (scaffolding). This library uses [Yeoman](https://yeoman.io/), however, in order to extend and reuse functionality we are using functions composition rather than Yeoman's usual approach of class inheritance. This will be covered in more detail in [How to Write a Generator](#how-to-write-a-generator) section. The code is generated from [EJS](https://ejs.co/) templates. 
+
+This library can be used as a standalone CLI tool, but most of the time it will be used from Studio. When used as a CLI tool it can interactively ask questions and use the **answers** to resolve interpolations in the templates. Studio will ask these questions using its graphical interface and invoke the generator CLI, passing the base64-encoded answers object as `--answers` **option**. There are other options, for example `--dest` that tells the generator where to put the generated files.  
+
+In addition to *answers* and *options* there is a **project model** - information about your Jmix project's entities, fetch plans, services, queries, etc. It can be obtained from Studio.
+
+> EJS template + options + answers + project model = generated code
+
+A **generator** is a combination of an EJS template and code that is responsible for asking questions and turning the answers, options and project model into generated code. For each **client** (e.g. React client, React Native client, etc.) there is always a generator that creates a starter app and zero or more generators that adds the components.
+
+> TIP: use `gen-cuba-front -h` to see the available clients, generators and options.
```

Comment:
I think we need rename `cuba` with `jmix`

Reply:
Yes, but let's do it as a separate ticket, there is a number of places where we need to change that.

Reply:
Agree.

Reply:
#84